hr_employee_document/models/ir_attachment.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+    _inherit = "ir.attachment"
+
+    @api.model
+    def _search(
```

Comment:
Security shouldn't be handled in `_search` method. This is a security hole, as you can browse directly some random records.